src/Shared/RoslynUtils/SymbolExtensions.cs

```diff
@@ -176,7 +177,27 @@ public static string GetDefaultValueString(this IParameterSymbol parameterSymbol
     {
         return !parameterSymbol.HasExplicitDefaultValue
             ? "null"
-            : SymbolDisplay.FormatLiteral((parameterSymbol.ExplicitDefaultValue ?? "null").ToString(), parameterSymbol.ExplicitDefaultValue is string);
+            : InnerGetDefaultValueString(parameterSymbol.ExplicitDefaultValue);
+    }
+
+    private static string InnerGetDefaultValueString(object? defaultValue)
+    {
+        return defaultValue switch
+        {
+            string s => SymbolDisplay.FormatLiteral(s, true),
+            char c => SymbolDisplay.FormatLiteral(c, true),
+            bool b => b ? "true" : "false",
+            null => "default",
+            float f when f is float.NegativeInfinity => "float.NegativeInfinity",
+            float f when f is float.PositiveInfinity => "float.PositiveInfinity",
+            float f when f is float.NaN => "float.NaN",
+            float f => $"{SymbolDisplay.FormatPrimitive(f, false, false)}F",
+            double d when d is double.NegativeInfinity => "double.NegativeInfinity",
+            double d when d is double.PositiveInfinity => "double.PositiveInfinity",
+            double d when d is double.NaN => "double.NaN",
```

Comment:
Does `double` get handled by `_ => SymbolDisplay.FormatPrimitive`?

Did that API not handle `double.NaN`, `float.PositiveInfinity`, etc.? It claims to support all the types, so kinda odd that they need to be special cased?

Reply:
Yep, the `double` is handled correctly by our terminal case. The comment on this API is rather deceptive given the conversation that is occuring in https://github.com/dotnet/roslyn/issues/58705. Evidently, `SymbolDisplay.FormatX` APIs are not intended for code gen. The recommended alternative [defined here](https://github.com/dotnet/roslyn/blob/0caa880a161057559d1dcaad890e4053f520136e/src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs#L1803) is defined in the workspace layer and would require introducing some new dependencies for the generator.

Given that we've been successful using SymbolDisplay in other places (displaying FQNs) I'm incline to workaround its limitations given the finite set of types we have to deal with here.

Reply:
The runtime does it like this: https://github.com/dotnet/runtime/pull/93160/files#diff-60e086268f0958325d865a75b553a6ed42b402d0a1792132e426a598172dace5R57

Reply:
> The runtime does it like this: https://github.com/dotnet/runtime/pull/93160/files#diff-60e086268f0958325d865a75b553a6ed42b402d0a1792132e426a598172dace5R57

Yeah, I think using `FormatPrimitive` is a sufficient alternative to calling the converter here. In any case, I'm planning on merging this to 9.0, evaluating the fix, then putting it into a servicing release of 8.0